### DISCUSSION TITLE
在setup中使用jsx返回时，devtools就无法监听查看数据了

### DISCUSSION THREAD

**Discussion by tangjian1891** [element-plus/element-plus#3878] [Q&A] [answered]
请问一下在开发时使用jsx还想在devtools查看调试数据，应该怎样? [#1295](https://github.com/vuejs/devtools/issues/1295)

`<script lang="tsx">
import { defineComponent, ref } from "vue-demi";
export default defineComponent({
  setup() {
    const countRef = ref(1);
    function foo() {
      countRef.value++;
    }
    return () => {
      return (
        <div>
          <div>{countRef.value}</div>
          <button onClick={foo}>增加</button>
        </div>
      );
    };
  },
});
</script>`

**Comment by jw-foss** [comment]
Try `expose` method

```ts
setup(props, { expose }) {
  const anyRef = ref()

  expose({
      anyRef,
  })
}
```

**Reply by tangjian1891** [reply]
> Try `expose` method
> 
> ```ts
> setup(props, { expose }) {
>   const anyRef = ref()
> 
>   expose({
>       anyRef,
>   })
> }
> ```

 测试了一下，expose是将组将暴露出，供实例调用，但是我这边devtools中仍然无法查看到暴露的数据。

**Reply by jw-foss** [reply]
You might need to raise the question in `vue` channel, we cannot offer help from here for the detailed requirement. Sorry

**Comment by sxzz** [comment, marked as answer]
https://github.com/vuejs/devtools/issues/1295

**Reply by tangjian1891** [reply]
> [vuejs/devtools#1295](https://github.com/vuejs/devtools/issues/1295)

这个我之前看过了，但是devtools并没有拿出好的方案。 所以我就很好奇，咱们Element-plus开发的时候，使用jsx的时候，也不需要devtools吗?所以就来这边咨询一下，使用vue3做jsx开发，调试数据咋调试的。

**Reply by sxzz** [reply]
我们用render函数的组件本身也是比较少的。我个人一般很少用到 devtools

**Reply by tangjian1891** [reply]
一般写业务比较依赖devtools,写通用组件确实还好。 感谢答疑，这个Q&A可以关掉了！
